bw_simapro_csv/blocks/process.py

```diff
@@ -0,0 +1,166 @@
+    "Emissions to soil",
+    "Emissions to water",
+    "Final waste flows",
+    "Materials/fuels",
+    "Non material emissions",
+    "Products",
+    "Resources",
+    "Social issues",
+    "Waste to treatment",
+    "Waste treatment",
+}
+HAS_SUBCATEGORY = {
+    "Economic issues"
+    "Emissions to air",
+    "Emissions to soil",
+    "Emissions to water",
+    "Final waste flows",
+    "Non material emissions",
+    "Resources",
+    "Social issues",
+}
+NO_SUBCATEGORY = {
+    "Avoided products",    
+    "Electricity/heat",
+    "Materials/fuels",
+    "Products",
+    "Waste to treatment",
+}
+PARAMETERS = {
+    "Calculated parameters",
+    "Input parameters",
+}
+
+
+# Exclude `e` for exponent
+has_letters = re.compile("[a-df-zA-CF-Z]+")
+has_numbers = re.compile("[0-9]+")
+
+
+class Process(SimaProCSVBlock):
+    """A life cycle inventory process, with inputs, products, and elementary exchanges"""
+
+    def __init__(self, block: List[list], header: dict):
+        self.parsed = {"metadata": {}}
+        self.raw = {}
+        self.index = 0
+
+        while not any(block[self.index]):
+            self.index += 1
+
+        # Start with metadata. This is stored as:
+        # Key
+        # Value
+        # On separate lines. Also, sometimes Value is missing.
+        while block[self.index][0] not in LIST_ELEMENTS:
+            k, v = self.pull_metadata_pair(block, header)
+            self.parsed["metadata"][k] = v
+
+        # These sections need access to the global variable store
+        # before they can be resolved
+        while self.index < len(block):
+            k, v = self.pull_raw_section(block, header)
+            self.raw[k] = v
+
+    def pull_raw_section(self, block: List[list], header: dict) -> (str, list):
+        """
+        0. name
+        1. subcategory
+        2. unit (or value)
+        3. value or formula (or unit)
+        4. uncertainty type
+        5. uncert. param.
+        6. uncert. param.
+        7. uncert. param.
+        8. comment
+
+        However, sometimes the value is in index 2, and the unit in index 3. Because why not! We assume default ordering unless we find a number in index 2.
+        """        
+        key = block[self.index][0]
+        data = []
+
+        self.index += 1
+
+        while any(block[self.index]):
+            if key in HAS_SUBCATEGORY:
+                data.append({
+                    'name': a,
+                    'categories': (key, b),
+                    'maybe_unit': c,
+                    'maybe_value': d,
+                    'kind': e,
+                    'field1': f,
+                    'field2': g,
+                    'field3': h
+                    } for (a,b,c,d,e,f,g,h) in block[self.index]
+                )
+            self.index += 1
+
+        # Skip empty line ending this section
+        self.index += 1
+        
+        return key, data
+
+    def resolve_unit_amount(self, a: str, b: str, key: str) -> dict:
```

Comment:
Ignore this, it should be consistent within the file so we will scan until we find an unambiguous case.